pkg/skr/gcpvpcpeering/updateStatus.go

```diff
@@ -55,5 +55,16 @@ func updateStatus(ctx context.Context, st composed.State) (error, context.Contex
 			Run(ctx, state)
 	}
 
+	if state.KcpVpcPeering.Status.Conditions[len(state.KcpVpcPeering.Status.Conditions)-1].LastTransitionTime.Time.After(state.ObjAsGcpVpcPeering().Status.Conditions[len(state.ObjAsGcpVpcPeering().Status.Conditions)-1].LastTransitionTime.Time) &&
+		state.KcpVpcPeering.Status.Conditions[len(state.KcpVpcPeering.Status.Conditions)-1].Message != state.ObjAsGcpVpcPeering().Status.Conditions[len(state.ObjAsGcpVpcPeering().Status.Conditions)-1].Message &&
+		kcpCondReady != nil &&
+		skrCondReady != nil {
+		return composed.UpdateStatus(state.ObjAsGcpVpcPeering()).
+			SetCondition(state.KcpVpcPeering.Status.Conditions[len(state.KcpVpcPeering.Status.Conditions)-1]).
```

Comment:
In general, we only keep one condition in other CRs. Is VPC Peering different? Doesn't that cause inconsistency? Maybe you can use 0 instead of len(xxx -1)?
Also what is the case that SKR has a more recent condition than KCP? Do you set conditions in SKR that are not coming from KCP? If all are coming from KCP then KCP is source of through anyway. 

Could you please add a comment to the code to explain this block?

BTW, we use mostly SetExclusiveConditions instead of SetCondition to make sure all other conditions are removed when we set a condition to keep only 1 condition.